pkg/controllers/user/helm/common/common.go

```diff
@@ -111,6 +108,48 @@ func InstallCharts(rootDir, port string, obj *v3.App) error {
 	return nil
 }
 
+func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
+	setValues := []string{}
+	// a user-supplied values file will overridden default values.yaml
+	if app.Spec.ValuesYaml != "" {
+		valuesYaml := filepath.Join(tempDir, "answers.yaml")
+		if err := ioutil.WriteFile(valuesYaml, []byte(app.Spec.ValuesYaml), 0755); err != nil {
+			return setValues, err
+		}
+		setValues = append([]string{"--values"}, valuesYaml)
+	}
+	// user-supplied values file will be overridden by --set parameters
+	if app.Spec.Answers != nil {
+		answers := app.Spec.Answers
+		var values, strValues []string
+		for k, v := range answers {
+			values, strValues = rancherTypedVal(k, v, values, strValues)
+		}
+		setValues = append([]string{"--set"}, strings.Join(values, ","))
+		setValues = append(setValues, "--set-string", strings.Join(strValues, ","))
+	}
+	return setValues, nil
+}
+
+// rancherTypedVal parse value based on prefix like str:
+func rancherTypedVal(key, value string, setValues, setStrValues []string) ([]string, []string) {
+	parts := strings.SplitN(value, ":", 2)
+	if len(parts) == 2 {
+		t := parts[0]
+		switch t {
+		// we only need to deal the str: prefix for now(e.g str:11111 should be convert to string `11111` instead of int),
+		// for the other types like bool, int will be passed properly by the original `helm --set`
+		case "str":
```

Comment:
What if I need something exactly like `str:...`? And I will never get what I want because `str:` will always be trimmed.

Reply:
This is true that user will never get the exact string value if its prefix is using the `str:`, and that's why we need to pick some unique but meaningful type of string as the prefix of rancher type values, in Helm they use `!!string 11111` to specify absolute string values for value like `11111`.
We can have more discussion on this with @alena1108 and @cjellick to conclude.

Reply:
move out and will add in another PR for rancher typed values.